components/TARGET_PSA/TARGET_TFM/COMPONENT_NSPE/interface/src/tfm_ns_lock_rtx.c

```diff
@@ -42,6 +42,7 @@ static const osMutexAttr_t ns_lock_attrib = {
 /**
  * \brief NS world, NS lock based dispatcher
  */
+__attribute__((weak))
```

Comment:
You should use MBED_WEAK instead.

```
/** MBED_WEAK
 *  Mark a function as being weak.
 *
 *  @note
 *  Functions should only be marked as weak in the source file. The header file
 *  should contain a regular function declaration to insure the function is emitted.
 *  A function marked weak will not be emitted if an alternative non-weak
 *  implementation is defined.
 *
 *  @note
 *  Weak functions are not friendly to making code re-usable, as they can only
 *  be overridden once (and if they are multiply overridden the linker will emit
 *  no warning). You should not normally use weak symbols as part of the API to
 *  re-usable modules.
 *
 *  @code
 *  #include "mbed_toolchain.h"
 *
 *  MBED_WEAK void foo() {
 *      // a weak implementation of foo that can be overriden by a definition
 *      // without  __weak
 *  }
 *  @endcode
 */
#ifndef MBED_WEAK
#if defined(__ICCARM__)
#define MBED_WEAK __weak
#elif defined(__MINGW32__)
#define MBED_WEAK
#else
#define MBED_WEAK __attribute__((weak))
#endif
#endif
```
- platform/mbed_toolchain.h

Reply:
The change to this file is temporary for my porting. Its final modification should come from TF-M repo (see https://github.com/ARMmbed/mbed-os/pull/10959#issuecomment-543799527).

Reply:
@VeijoPesonen I still use `__attribute__((weak))` because all toolchains support it. This is also what https://review.trustedfirmware.org/c/trusted-firmware-m/+/1460 is adopting.